### PR TITLE
Standardize jump links

### PR DESCRIPTION
**Pull request by anselmbradford**
Brings in design-system jump link updates from:

https://github.com/cfpb/design-system/pull/1987
https://github.com/cfpb/design-system/pull/1988
https://github.com/cfpb/design-system/pull/1989
https://github.com/cfpb/design-system/pull/1990
https://github.com/cfpb/design-system/pull/1991

## Changes

- Deprecate `m-list__link` to `a-link a-link--jump`.
- Fix issue where jump links in a stack would not highlight the top border on hover.

## How to test this PR

1. Example pages are below. The majority should have jump links (search for `a-link--jump` in the source).

KBYO
http://localhost:8000/know-before-you-owe/
http://localhost:8000/know-before-you-owe/compare/
http://localhost:8000/know-before-you-owe/timeline/

Careers - should not have a broken subtext
http://localhost:8000/about-us/careers/
http://localhost:8000/about-us/careers/supervisory-investigator-24-cfpb-139-p24-cfpb-138-mp/
http://localhost:8000/about-us/careers/application-process/

Financial Well-being
http://localhost:8000/consumer-tools/financial-well-being/
http://localhost:8000/consumer-tools/financial-well-being/about/
http://localhost:8000/consumer-tools/financial-well-being/results/

Ask CFPB 
http://localhost:8000/ask-cfpb/
http://localhost:8000/ask-cfpb/do-buy-now-pay-later-bnpl-loans-have-fees-en-2118/

D&I
http://localhost:8000/about-us/diversity-and-inclusion/voluntary-assessment-onboarding-form/
http://localhost:8000/about-us/diversity-and-inclusion/voluntary-assessment-onboarding-form/privacy-act-statement/

TDP
http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/bouncing-ball-money-choices/

Retirement
http://localhost:8000/es/herramientas-del-consumidor/jubilacion/antes-de-solicitar/

OAH/BAH Rate Explorer
http://localhost:8000/owning-a-home/explore-rates/

FAHC
http://localhost:8000/find-a-housing-counselor/

ROUT
http://localhost:8000/rural-or-underserved-tool/

iRegs
http://localhost:8000/rules-policy/regulations/1002/2023-04-19/interp-1/

FIG
http://localhost:8000/data-research/small-business-lending/filing-instructions-guide/2024-guide/

Credit Card agreements database related links - sidebar related links should NOT be jump link at mobile
http://localhost:8000/credit-cards/agreements/issuer/1st-advantage-federal-credit-union/

Info Unit Groups
http://localhost:8000/about-us/

Featured Content
http://localhost:8000/consumer-tools/credit-reports-and-scores/consumer-reporting-companies/

Notification
http://localhost:8000/arbitration-rule/

Text Introduction
http://localhost:8000/administrative-adjudication-proceedings/administrative-adjudication-docket/

Homepage - footer links should be unchanged.
http://localhost:8000/

## Notes and todos

The new behavior of jump links vertically centers the external link icon after the text, so if the link is very long, it'll end up on the right-hand side. We should probably encourage shorter links if that's an issue. I think it improves cases like this though:

Before:
Screenshot 2024-05-31 at 5 25 29 PM

After:
Screenshot 2024-05-31 at 5 25 38 PM

### CODE REVIEW

File: cfgov/unprocessed/css/enhancements/layout-1-3.less [hunk 1]
```diff
@@ -120,6 +120,10 @@
     top: 0;
     align-self: start;
 
+    // This puts the secondary nav behind the mega menu hamburger menu,
+    // but above other things on the page.
+    z-index: 9;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/layout-1-3.less:125]: @contolini is this change for FIG sticky nav OK?

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html [hunk 2]
```diff
@@ -136,10 +136,10 @@ <h4>Activity characteristics</h4>
         <h4 class="u-mb5">National standards</h4>
         <p class="u-mb20"><small>
             (<a
-            class="a-link a-link--icon" target="_blank" rel="noreferrer noopener"
+            class="a-link" target="_blank" rel="noreferrer noopener"
             href="https://files.consumerfinance.gov/f/documents/2021_Natl_Standards_Downloadable_final.pdf"
-            ><span class="a-link__text">What are the national standards? </span>
-            <span class="a-icon">{{ svg_icon('download') }}</span></a>)
+            ><span class="a-link__text">What are the national standards?</span
+            ><span class="a-icon"> {{ svg_icon('download') }}</span></a>)
```
Comment (wpears) [inline cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html:142]:
```suggestion
            ><span class="a-icon">{{ svg_icon('download') }}</span></a>)
```

**Review by wpears — CHANGES_REQUESTED** [review]

File: cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html (hunk 2 above)
Comment (anselmbradford) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html:142]: Sooooooo this space is actually intentional. 

No space:
Screenshot 2024-06-04 at 1 04 05 PM

Space:
Screenshot 2024-06-04 at 1 04 40 PM

Comment (wpears) [reply cfgov/teachers_digital_platform/jinja2/teachers_digital_platform/activity_page.html:142]: Hah! I approve while disapproving!

**Review by wpears — APPROVED** [review]

**Pull request merged**